src/test/regress/expected/multi_follower_select_statements.out

```diff
@@ -0,0 +1,105 @@
+\c - - - :master_port
+-- do some setup
+SELECT 1 FROM master_add_node('localhost', :worker_1_port);
+ ?column? 
+----------
+        1
+(1 row)
+
+SELECT 1 FROM master_add_node('localhost', :worker_2_port);
+ ?column? 
+----------
+        1
+(1 row)
+
+CREATE TABLE the_table (a int, b int);
+SELECT create_distributed_table('the_table', 'a');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+INSERT INTO the_table (a, b) VALUES (1, 1);
+INSERT INTO the_table (a, b) VALUES (1, 2);
+-- connect to the follower and check that a simple select query works, the follower
+-- is still in the default cluster and will send queries to the primary nodes
+\c - - - :follower_master_port
+SELECT * FROM the_table;
+ a | b 
+---+---
+ 1 | 1
+ 1 | 2
+(2 rows)
+
+-- now, connect to the follower but tell it to use secondary nodes. There are no
+-- secondary nodes so this should fail.
+-- (this is :follower_master_port but substitution doesn't work here)
+\c "port=57700 dbname=regression options='-c\ citus.use_secondary_nodes=always'"
+SELECT * FROM the_table;
+ERROR:  node group 2 does not have a secondary node
+-- add the secondary nodes and try again, the SELECT statement should work this time
+\c - - - :master_port
+SELECT 1 FROM master_add_node('localhost', :follower_worker_1_port,
+  groupid => (SELECT groupid FROM pg_dist_node WHERE nodeport = :worker_1_port),
+  noderole => 'secondary');
+ ?column? 
+----------
+        1
+(1 row)
+
+SELECT 1 FROM master_add_node('localhost', :follower_worker_2_port,
+  groupid => (SELECT groupid FROM pg_dist_node WHERE nodeport = :worker_2_port),
+  noderole => 'secondary');
+ ?column? 
+----------
+        1
+(1 row)
+
+-- (the previous \c is a good example of how to change a variable temporarily and left
+-- in for future reference but from now on we always want the follower to read from
+-- secondaries so permanently change it. use_secondary_nodes is SU_BACKEND so you
+-- must reconnect in order for the change to take effect)
+\c - - - :follower_master_port
+ALTER SYSTEM SET citus.use_secondary_nodes TO 'always';
+SELECT pg_reload_conf();
+ pg_reload_conf 
+----------------
+ t
+(1 row)
+
+\c - - - -
+-- now that we've added secondaries this should work
+SELECT * FROM the_table;
+ a | b 
+---+---
+ 1 | 1
+ 1 | 2
+(2 rows)
+
+-- okay, now let's play with nodecluster. If we change the cluster of our follower node
+-- queries should stat failing again, since there are no worker nodes in the new cluster
+-- cluster_name is also SU_BACKEND, so we need to reconnect after changing it
+ALTER SYSTEM SET citus.cluster_name TO 'second-cluster';
+SELECT pg_reload_conf();
+ pg_reload_conf 
+----------------
+ t
+(1 row)
+
+\c - - - -
+-- there are no secondary nodes in this cluster, so this should fail!
+SELECT * FROM the_table;
+ERROR:  the metadata is inconsistent
+DETAIL:  there is a placement in group 2 but there are no nodes in that group
```

Comment:
In case the user ever gets here, could we move the detail into the error? Details don't show up in many ORMs.

Also, "group 2" -> "node group 2"